app/Testing/Context/MailSending.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brewmap\Testing\Context;
+
+use Behat\Behat\Context\Context;
+use Brewmap\Notifications\Mail\EmailChangeNotification;
+use Illuminate\Notifications\AnonymousNotifiable;
+use Illuminate\Support\Facades\Notification;
+
+class MailSending implements Context
+{
+    public function __construct()
+    {
+        Notification::fake();
+    }
+
+    /**
+     * @Then an email should be sent
+     */
+    public function anEmailShouldBeSent(): void
+    {
+        Notification::assertSentTo(new AnonymousNotifiable(), EmailChangeNotification::class, function (EmailChangeNotification $notification, array $channels, AnonymousNotifiable $notifiable) {
```

Comment:
You can use arrow function for that:
```suggestion
       Notification::assertSentTo(new AnonymousNotifiable(), EmailChangeNotification::class, fn (EmailChangeNotification $notification, array $channels, AnonymousNotifiable $notifiable): bool => $notifiable->routes["mail"] === "new_email@example.com");
```

You forgot about return type. You didn't remove facade too.

Reply:
In this case with facade, I am waiting for help from Mateusz.

Reply:
It was failing with `550 relaying denied` message, probably because it was still trying to use real notification dispatcher in tests.

Nothing was mocked when you just simply instantiated `NotificationFake` and tried to perform assertions.

You need to swap manually default implementation of  `Illuminate\Contracts\Notifications\Dispatcher` somewhere in tests.

You can do it context's constructor eg.
```php
$this->notificatonFake = new NotificationFake();

app()->bind(Dispatcher::class, function ($app) {
       return $this->notificatonFake;
});

```

so then you could use it in steps eg.
```php
/**
* @Then an email should be sent
*/
public function anEmailShouldBeSent(): void
{
      $this->notificatonFake->assertSentTo(.....)
}
```

or you could create new `TestServiceProvider` where you could bind `NotificationFake` the same way

Reply:
okey, i will try with it.

Reply:
it works, thank you so much.

Reply:
`TestingServiceProvider` with binding mocks is not a good option. Soon it will be huge and it will mock a milion classes for every test.